treewide: move to libiio 1.0i API

nunojsa opened the pull request:
While at it, increased cmake minimum version to 3.10 as in other projects.

@pcercuei, your review would be valuable in here. Note that I'm only moving from one API to another. Not code refactor is intended at all (only the cmake stuff to avoid an annoying warning)

@tfcollins this is only compile tested so it would be great if you or someone else could give this a test.

nunojsa committed:
treewide: move to libiio 1.0i API

While at it, increased cmake minimum version to 3.10 as in other
projects.

Signed-off-by: Nuno Sa <nuno.sa@analog.com>

pcercuei commented:
I'd hold that for now, I plan to rework how the attributes are accessed in Libiio v1.0 - which should be ready by the end of next week.
(I'm on vacation this week FYI)

nunojsa commented:
> I'd hold that for now, I plan to rework how the attributes are accessed in Libiio v1.0 - which should be ready by the end of next week.
(I'm on vacation this week FYI)

Alright, I'm also not sure what's the plan here (I see CI also needs updating when moving to 1.0).

The background for the PR is that meta-adi main branch follows the other projects main branch. Therefore, all projects (this one and the adrv9009-monitor) that depend on libiio are now failing to build. Anyways, I guess it might takes some time for the dust to settle, so I might just temporarily point the libiio recipe to the `libiio-v0` branch.

tfcollins commented:
@tagoylo can you check the hardware harness to make sure this runs without issue

tagoylo commented:
Following errors were encountered when after running make

> [100%] Linking C executable FMComms5SyncTest
> /usr/bin/ld: CMakeFiles/FMComms5SyncTest.dir/fmcomms5_sync_test.c.o: in function 'check_phase_sma':
> fmcomms5_sync_test.c:(.text+0x8d8): undefined reference to 'iio_create_channels_jmask'
> collect2: error: ld returned 1 exit status
> make[2]: *** [test/CMakeFiles/FMComms5SyncTest.dir/build.make:86: test/FMComms5SyncTest] Error 1
> make[1]: *** [CMakeFiles/Makefile2:288: test/CMakeFiles/FMComms5SyncTest.dir/all] Error 2
> make: *** [Makefile:163: all] Error 2
> script returned exit code 2


Full log: 
[libad9361-make-error.txt](https://github.com/analogdevicesinc/libad9361-iio/files/12581822/libad9361-make-error.txt)

libiio branch used: main

nunojsa commented:
> Full log:
libad9361-make-error.txt

> libiio branch used: main

It looks like i added some typo right before the pull... will update soon

tfcollins commented:
> @tfcollins I think @ccraluca will be out for a few months at least, so somebody else should do it.

@SRaus who can look at the Azure pieces?

AAndrisa commented:
I created a branch from 'staging/libiio1-support' called  'staging/libiio1-artifacts' and made changes so that the artifacts are now taken from the main branch of libiio but it seems that are still some build issues. Or should this change still use artifacts from libiio-v0 branch?

pcercuei commented:
@SRaus the API of Libiio in the `main` branch won't be stable until the v1.0 release is tagged.
So I'd advise against updating anything to the will-be-v1.0 API (unless it's for testing purpose - then yes please) as it's bound to be broken at some point.

tfcollins commented:
> @SRaus the API of Libiio in the `main` branch won't be stable until the v1.0 release is tagged. So I'd advise against updating anything to the will-be-v1.0 API (unless it's for testing purpose - then yes please) as it's bound to be broken at some point.

This would probably be a useful canary build to have though. I think we have the following options:
1. Support both with feature flags
2. Merge this with some CMake updates to help with versioning checks if possible. We really need a true CMake config in libiio and libad9361 that allows for the use of `find_package` anyway
3. Move to a stable/next_stable/main type branching strategy first

SRaus commented:
> the API of Libiio in the `main` branch won't be stable until the v1.0 release is tagged
In this case, from my point of view, this means that it shouldn't be in master yet. 
But instead, the changes should be in a dev branch, manually build and test libad9361 (and all other components that use libiio) with artifacts from that specific dev branch. Finally, when everything was clear, to push to master/main all PR's from all dependent repos on the same time.

SRaus commented:
> This would probably be a useful canary build to have though. I think we have the following options:
> 
> 1. Support both with feature flags
> 2. Merge this with some CMake updates to help with versioning checks if possible. We really need a true CMake config in libiio and libad9361 that allows for the use of `find_package` anyway
> 3. Move to a stable/next_stable/main type branching strategy first

In this case, I would go with #2 , definitely not touching next_stable (which we are trying to stabilize) or last_stable.

pcercuei commented:
> > the API of Libiio in the `main` branch won't be stable until the v1.0 release is tagged
> 
> In this case, from my point of view, this means that it shouldn't be in master yet. But instead, the changes should be in a dev branch, manually build and test libad9361 (and all other components that use libiio) with artifacts from that specific dev branch. Finally, when everything was clear, to push to master/main all PR's from all dependent repos on the same time.

The "master" branch (now "main") has always been a **development** branch. If you want a stable experience, use the releases. And I did work on a separate "dev" branch for about two years before it was eventually merged back into "master". Did you test libad9361 vs. the dev branch then?

SRaus commented:
> > > the API of Libiio in the `main` branch won't be stable until the v1.0 release is tagged
> > 
> > 
> > In this case, from my point of view, this means that it shouldn't be in master yet. But instead, the changes should be in a dev branch, manually build and test libad9361 (and all other components that use libiio) with artifacts from that specific dev branch. Finally, when everything was clear, to push to master/main all PR's from all dependent repos on the same time.
> 
> The "master" branch (now "main") has always been a **development** branch. If you want a stable experience, use the releases. And I did work on a separate "dev" branch for about two years before it was eventually merged back into "master". Did you test libad9361 vs. the dev branch then?

Haven't tested libad9361 with dependencies from dev branch - I'm expecting that beside testing, libad repos are requiring some changes in order to work.
 
Regarding libiio releases - I agree that they are stable and we can rely on them.
But can we build/test latest libad9361 with artifacts from latest libiio release? (Probably yes right now, since latest release is pretty recent / and probably no one month ago, before creating v0.25, since previous release would have been too old - more than one year old).

ekigwana committed:
treewide: update move to libiio 1.0i API

Signed-off-by: Edward Kigwana <ekigwana@gmail.com>

AAndrisa committed:
azure-pipelines.yaml: update arm dependecies

qemu is not supported and qemu-user-static needs to be installed
instead.

Based on this discussion:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1087822
Disable Address Space Layout Randomization for the ppc64le, x390x,
armv7/8 builds to pass.

Signed-off-by: Andreea Andrisan <Andreea.Andrisan@analog.com>

AAndrisa committed:
CI:travis: disable pytest

There is no support of pytests fro libiio v1 meaning this step will
fail. Until the propper support will be added we can disable this step.

Signed-off-by: Andreea Andrisan <Andreea.Andrisan@analog.com>

AAndrisa committed:
CI: update naming for libiio v1 version

The naming of libiio.lib was changed to libiio1.lib

Signed-off-by: Andreea Andrisan <Andreea.Andrisan@analog.com>

999Marius committed:
fmcomms5: port streaming to libiio v1 multi-buffer API

iio_device_create_buffer() and iio_buffer_destroy() are gone in libiio v1.
The buffer is got with iio_device_get_buffer().
iio_buffer_create_stream() now takes the channels mask directly.

Signed-off-by: Marius Lucacel <marius.lucacel@analog.com>
